packages/asset-buyer/src/utils/calculate_liquidity.ts

```diff
@@ -0,0 +1,34 @@
+import { BigNumber } from '@0x/utils';
+
+import { LiquidityForAssetData, OrdersAndFillableAmounts } from '../types';
+
+import { orderUtils } from './order_utils';
+
+export const calculateLiquidity = (ordersAndFillableAmounts: OrdersAndFillableAmounts): LiquidityForAssetData => {
+    const { orders, remainingFillableMakerAssetAmounts } = ordersAndFillableAmounts;
+    const liquidityInBigNumbers = orders.reduce(
+        (acc, order, curIndex) => {
+            const availableMakerAssetAmount = remainingFillableMakerAssetAmounts[curIndex];
+            if (availableMakerAssetAmount === undefined) {
+                throw new Error(`No corresponding fillableMakerAssetAmounts at index ${curIndex}`);
+            }
+
+            const tokensAvailableForCurrentOrder = availableMakerAssetAmount;
+            const ethValueAvailableForCurrentOrder = orderUtils.getTakerFillAmount(order, availableMakerAssetAmount);
+            return {
+                tokensAvailableInBaseUnits: acc.tokensAvailableInBaseUnits.plus(tokensAvailableForCurrentOrder),
+                ethValueAvailableInWei: acc.ethValueAvailableInWei.plus(ethValueAvailableForCurrentOrder),
+            };
+        },
+        {
+            tokensAvailableInBaseUnits: new BigNumber(0),
+            ethValueAvailableInWei: new BigNumber(0),
+        },
+    );
+
+    // Turn into regular numbers
+    return {
+        tokensAvailableInBaseUnits: liquidityInBigNumbers.tokensAvailableInBaseUnits.toNumber(),
+        ethValueAvailableInWei: liquidityInBigNumbers.ethValueAvailableInWei.toNumber(),
+    };
```

Comment:
hmmm would've been nice to stick to `BigNumber`, but yea the === 0 case is not great. 

```
new BigNumber(2) > 1 
true
new BigNumber(1) > 1 
false
new BigNumber(0) > 1 
false
new BigNumber(0) === 0 
false
new BigNumber(0) == 0 
true
```
Maybe keep asset_buyer using `BigNumber` and convert it to number at the Instant level?

Reply:
I opted in for using regular numbers here, because I didn't want the integrator to have to introduce the BigNumber dependency for checking liquidity

Reply:
Ah, good point!  Here's a few of the ideas we discussed:

**Return an instance of BigNumber**. Export zeroExInstant.BigNumber on the window.  To ensure that integrators don't mistake this for a regular number, prepend "bigNumber" onto the return value key names

_Pros:_
- Guarantees precision
- We return BigNumbers throughout the rest of the code-base, so this is likely what a 0x-er would expect

_Cons_
- Forces us to always ship a version of BigNumber with our bundle, increasing the minimum bundle size in a future where we code-split
- If they also package their own version of BigNumber, BigNumber code could be shipped twice the browser
- Could get into weird scenarios if they use a static method (i.e. `BigNumber.maximum`) using a different version of BigNumber

**Return a string representation of the number**:  Have `asset-buyer` return a BigNumber, but call `.toString()` on it in the helper method on the `zeroExInstant` window object.  Prepend the word "string" onto the return value keys so folks do not mistake it for a number. 
 This would allow integrator to use whatever number library they'd like to parse the value.  They could opt-in to use something like [big.js](https://github.com/MikeMcl/big.js/) instead of BigNumber if they were concerned about bundle size (this also would require us updating the check liquidity method to not rely on asset-buyer and BigNumber)

**Rip out the liquidity # response all together**: Just return a boolean representing if the token has liquidity.  This would handle the "Should I show the 'Buy' button for a specific erc20 token?" question, and would allow the integrator to query a list of relayers and choose the first one that has liquidity.  This _does not_ handle the situation in which there more than one relayer supports a token and an integrator wants to choose the relayer with the most liquidity

With instant being served by a cdn which always hosts the latest version of the code, we don't want to later change this interface and break integrations, so it seems important to get this right.